community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreAbstractNodeCursor.java

```diff
@@ -134,16 +140,29 @@ public long id()
         return singleLabelCursor.get().init( parseLabelsField( nodeRecord ).get( neoStores.getNodeStore() ), labelId );
     }
 
+    private Lock shortLivedReadLock()
+    {
+        Lock lock = lockService.acquireNodeLock( nodeRecord.getId(), LockService.LockType.READ_LOCK );
+        if ( lockService != NO_LOCK_SERVICE )
+        {
+            // It's safer to re-read the node record here, specifically nextProp, after acquiring the lock
+            neoStores.getNodeStore().loadRecord( nodeRecord.getId(), nodeRecord );
```

Comment:
Could this fail? Perhaps wrap in a try/catch where the catch-block releases the lock and re-throws?